features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64LRUCache.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.content.ComponentCallbacks2
+import android.content.res.Configuration
+import android.graphics.drawable.Drawable
+import android.util.LruCache
+import androidx.annotation.VisibleForTesting
+
+@Suppress("MagicNumber")
+internal object Base64LRUCache : Cache, ComponentCallbacks2 {
+    @Suppress("UnsafeThirdPartyFunctionCall") // maxMemory does not throw
+    private val CACHE_MEMORY_SIZE = (Runtime.getRuntime().maxMemory() / 8).toInt()
+    private val ON_LOW_MEMORY_SIZE = CACHE_MEMORY_SIZE / 2 // 50% size
+    private val ON_MODERATE_MEMORY_SIZE = (CACHE_MEMORY_SIZE / 4) * 3 // 75% size
+
+    private var cache: LruCache<Int, String> = object : LruCache<Int, String>(CACHE_MEMORY_SIZE) {
+        override fun sizeOf(key: Int?, value: String): Int {
+            return value.toByteArray().size
+        }
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {}
+
+    override fun onLowMemory() {
+        cache.evictAll()
+    }
+
+    @VisibleForTesting
+    internal fun setBackingCache(cache: LruCache<Int, String>) {
```

Comment:
minor: maybe just a provider `(Int) -> LruCache` in the constructor with default value for prod and mock value for tests?